hash-tables/overview-of-hash-tables.md

```diff
@@ -0,0 +1,153 @@
+# Overview of Hash Tables
+
+## Learning Goals
+
+- Define hash table
+- Define hash function
+- Describe how hash tables resolve collisions
+
+## Introduction
+
+Python dictionaries are fascinating! They store key-value pairs, and, as long as we have the key, getting the value has a time complexity of O(1).
+
+This is in direct comparison to arrays/lists. We could store elements in an array, but finding specific entries in an array requires looping through potentially the entire list an O(n) operation.
+
+How do Python dictionaries achieve this speed? How do keys and values get allocated memory? Under the hood, Python dictionaries are _hash tables_.
+
+When we consider how hash tables are designed and implemented, we will:
+
+- Practice algorithmic and design thinking around data structures
+- Consider how complex data structures are implemented
+- Learn powerful problem-solving techniques for problems often encountered in interviews
+- Appreciate dictionaries more!
+
+## Vocabulary and Synonyms
+
+| Vocab         | Definition                                                                                          | Synonyms         | How to Use in a Sentence                                                                                                                                                                                      |
+| ------------- | --------------------------------------------------------------------------------------------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Hash Table    | A data structure that implements an associative array, which maps unique key identifiers to values. | Hash Map         | "Hash tables have an O(1) lookup time."                                                                                                                                                                       |
+| Hash Function | A function used by a hash table to map a key to an index.                                           | Hashing function | "The hash function converts a key into an index in an associative array"                                                                                                                                      |
+| Collision     | When multiple keys map to the same element in a hash table's internal array.                        | -                | "The hash function wanted to insert a value into an index where a different value already existed, which is a collision. The hash table needs to handle this collision in order to keep both pieces of data." |
```

Comment:
the function doesn't want to do anything, it just returns a result. it has the effect of multiple pairs being inserted at the same location
```suggestion
| Collision     | When multiple keys map to the same element in a hash table's internal array.                        | -                | "The hash function result caused a value to be inserted into an index where a different value already existed, which is a collision. The hash table needs to handle this collision in order to keep both pieces of data." |
```